syntax = "proto3";

package book;

// Service definition
service BookService {
  rpc GetBook (BookRequest) returns (BookResponse);
  rpc ListBooks (Empty) returns (BookList);
  rpc UpdateStock (StockRequest) returns (StockResponse);
}

// Request for a single book
message BookRequest {
  int32 book_id = 1;
}

// Response for a single book
message BookResponse {
  int32 id = 1;
  string title = 2;
  string author = 3;
  int32 year = 4;
  string isbn = 5;
  int32 pages = 6;
  int32 stock = 7;
}

// Empty message for listing all books
message Empty {}

// List of books
message BookList {
  repeated BookResponse books = 1;
}

// Request to update stock
message StockRequest {
  int32 book_id = 1;
  int32 stock = 2;
}

// Response for stock update
message StockResponse {
  string message = 1;
}
